pkg/provider/azure_loadbalancer.go

```diff
@@ -3053,7 +3053,7 @@ func equalLoadBalancingRulePropertiesFormat(s *network.LoadBalancingRuleProperti
 	}
 
 	properties = properties && reflect.DeepEqual(s.FrontendIPConfiguration, t.FrontendIPConfiguration) &&
-		reflect.DeepEqual(s.BackendAddressPool, t.BackendAddressPool) &&
+		string.EqualFold(s.BackendAddressPool.id, t.BackendAddressPool.id) &&
```

Comment:
Could you add a util comparing function for `SubResource` type and then use it for BackendAddressPool, FrontendIPConfiguration and Probe?